### ISSUE TITLE
Change the name of primary package to leprechaun

### ISSUE THREAD

**Issue by icyflame** [icyflame/leprechaun#3]
- main package isn't documented by godoc
- use the primary package name as leprechaun and in `main.go` simply call `leprechaun.Main()`

**Comment by kshitij10496** [comment]
This is a good step forward.

However, I would suggest that we focus on separating the library code from the application rather than simply renaming everything to the `leprechaun` package.

**Comment by icyflame** [comment]
Yeah, absolutely! This is precisely what I found out during a set of mis-steps where I just naively tried to change the package name and hoped that everything would continue to work. 

![image](https://user-images.githubusercontent.com/3668034/35469429-9aac1368-035a-11e8-808c-13d086d59c51.png)

I am going to make "leprechaun" a library and make a simple "main.go" script which will act as the application and deploy that to Heroku.
